src/realm/__tests__/realmReducer-test.js

```diff
@@ -233,5 +234,59 @@ describe('realmReducer', () => {
         check(false, false);
       });
     });
+
+    describe('type `realm`, op `update_dict`', () => {
+      const eventCommon = { id: 0, type: EventTypes.realm, op: 'update_dict', property: 'default' };
+
+      const mkCheck = <S: $Keys<RealmState>, E: $Keys<RealmDataForUpdate>>(
+        statePropertyName: S,
+        eventPropertyName: E,
+      ): (($ElementType<RealmState, S>, $ElementType<RealmDataForUpdate, E>) => void) => (
+        initialStateValue,
+        eventValue,
+      ) => {
+        const initialState = { ...eg.plusReduxState.realm };
+        // $FlowFixMe[prop-missing]
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        initialState[statePropertyName] = initialStateValue;
+
+        const expectedState = { ...initialState };
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        expectedState[statePropertyName] = eventValue;
+
+        expect(
+          realmReducer(initialState, {
+            type: EVENT,
+            event: {
+              ...eventCommon,
+              // $FlowFixMe[invalid-computed-prop]
+              data: { [eventPropertyName]: eventValue },
+            },
+          }),
+        ).toEqual(expectedState);
+      };
+
+      describe('name / name', () => {
+        const check = mkCheck('name', 'name');
+        check('foo', 'foo');
+        check('foo', 'bar');
```

Comment:
It turns out that this bit of test code doesn't actually do anything! These tests don't run.

The same is true of the existing `twenty_four_hour_time` tests above.

You can witness this by running Jest directly, and narrowing it down to just the one file so that (by default) it gives more details:
```
$ npx jest realmReducer --selectProjects ios
Running one project: ios
 PASS   ios  src/realm/__tests__/realmReducer-test.js
  realmReducer
    REGISTER_COMPLETE
      ✓ updates as appropriate on a boring but representative REGISTER_COMPLETE (2 ms)
    ACCOUNT_SWITCH
      ✓ resets state
    EVENT_UPDATE_DISPLAY_SETTINGS
      ✓ change the display settings (1 ms)
    EVENT_REALM_EMOJI_UPDATE
      ✓ update state to new realm_emoji (1 ms)
    EVENT_REALM_FILTERS
      ✓ update state to new realm_filter

Test Suites: 1 passed, 1 total
Tests:       5 passed, 5 total
Snapshots:   0 total
Time:        3.239 s, estimated 4 s
Ran all test suites matching /realmReducer/i.
```

No mention of these test cases.

The issue is that there's no `test` call: this `describe` calls `check`, and `check` has an `expect` but no `test` around it.

Two reasonable solutions would be:
* Switch this inmost `describe` to a `test`: so `test('name / name', …`.
* Have `check` wrap its body in `test`. (Including the logic to construct the initial and expected states -- best to push work inside the `test` callbacks where possible, because that way Jest can get more quickly through all the outer logic that defines what test cases exist, and have its parallelization magic fully apply to that work.)

Reply:
> You can witness this by running Jest directly, and narrowing it down to just the one file so that (by default) it gives more details:

BTW this view also displays something I'd want a fix for if we pursued the more complex approach: the individual test cases get extremely long names. Here's a single one of them:
```
        ✓ 

initial state: {
  name: 'foo',
  description: 'foo',
  nonActiveUsers: [ [length]: 0 ],
  filters: [ [length]: 0 ],
  emoji: {},
  videoChatProvider: null,
  mandatoryTopics: false,
  messageContentDeleteLimitSeconds: null,
  messageContentEditLimitSeconds: 1,
  pushNotificationsEnabled: true,
  webPublicStreamsEnabled: false,
  createWebPublicStreamPolicy: 7,
  enableSpectatorAccess: false,
  canCreateStreams: true,
  isAdmin: false,
  isOwner: false,
  isModerator: false,
  user_id: undefined,
  email: undefined,
  crossRealmBots: [ [length]: 0 ],
  twentyFourHourTime: false
}

 event data: {
  name: 'foo',
  description: 'bar',
  enable_spectator_access: false,
  create_web_public_stream_policy: 6
}

 expected state: {
  name: 'foo',
  description: 'bar',
  nonActiveUsers: [ [length]: 0 ],
  filters: [ [length]: 0 ],
  emoji: {},
  videoChatProvider: null,
  mandatoryTopics: false,
  messageContentDeleteLimitSeconds: null,
  messageContentEditLimitSeconds: 1,
  pushNotificationsEnabled: true,
  webPublicStreamsEnabled: false,
  createWebPublicStreamPolicy: 6,
  enableSpectatorAccess: false,
  canCreateStreams: true,
  isAdmin: false,
  isOwner: false,
  isModerator: false,
  user_id: undefined,
  email: undefined,
  crossRealmBots: [ [length]: 0 ],
  twentyFourHourTime: false
}
```

Then there are about 700 of those, so together they spew a huge amount of output to the terminal.

Reply:
> It turns out that this bit of test code doesn't actually do anything! These tests don't run.

Oh wow! Thanks for catching that! 🤯

> BTW this view also displays something I'd want a fix for if we pursued the more complex approach: the individual test cases get extremely long names. Here's a single one of them:
> 
> …
> 
> Then there are about 700 of those, so together they spew a huge amount of output to the terminal.

Yeah, I noticed this problem but decided to check in before spending time trying to solve it (which would mean adding even more complexity).

Reply:
> * Have `check` wrap its body in `test`. (Including the logic to construct the initial and expected states -- best to push work inside the `test` callbacks where possible, because that way Jest can get more quickly through all the outer logic that defines what test cases exist, and have its parallelization magic fully apply to that work.)

I'll use this approach for my next revision; I like how it lets me give test names more granularly.